File: src/net/yu_yum/android/debug/YLog.java
package net.yu_yum.android.debug;

import jp.yumyum.BuildConfig;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * ログユーティリティクラス
 * 
 * @author yu
 *
 */
public class YLog extends BroadcastReceiver {

    private static final String TAG = "YLog";
    // インテントでログの有効／無効を制御可能（AndroidManifestに記述が必要）
    public static String ACTION_LOG_ENABLE = "net.yu_yum.LOG_ENABLE";
    public static String ACTION_LOG_DISABLE = "net.yu_yum.LOG_DISABLE";

    private static boolean sEnabled;
    static {
        sEnabled = BuildConfig.DEBUG;
    }

    public static void d(String tag, String msg) {
        if (sEnabled) {
            Log.d(tag, msg);
        }
    }

    public static void i(String tag, String msg) {
        if (sEnabled) {
            Log.i(tag, msg);
        }
    }

    public static void w(String tag, String msg) {
        if (sEnabled) {
            Log.w(tag, msg);
        }
    }

    public static void e(String tag, String msg) {
        if (sEnabled) {
            Log.e(tag, msg);
        }
    }

    /**
     * インテントでログの有効／無効を制御
     */
    @Override
    public void onReceive(Context arg0, Intent arg1) {
        String action = arg1.getAction();
        if (ACTION_LOG_ENABLE.equals(action)) {
            sEnabled = true;
            Log.d(TAG, "Log ENABLED");
        } else if (ACTION_LOG_DISABLE.equals(action)) {
            sEnabled = false;
            Log.d(TAG, "Log DISABLED");
        }
    }
}

File: app/src/main/java/net/yu_yum/android/debug/YLog.kt
package net.yu_yum.android.debug

import jp.yumyum.BuildConfig
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * ログユーティリティクラス
 *
 * @author yu
 */
class YLog : BroadcastReceiver() {

    /**
     * インテントでログの有効／無効を制御
     */
    override fun onReceive(arg0: Context, arg1: Intent) {
        val action = arg1.action
        if (ACTION_LOG_ENABLE == action) {
            sEnabled = true
            Log.d(TAG, "Log ENABLED")
        } else if (ACTION_LOG_DISABLE == action) {
            sEnabled = false
            Log.d(TAG, "Log DISABLED")
        }
    }

    companion object {

        private val TAG = "YLog"
        // インテントでログの有効／無効を制御可能（AndroidManifestに記述が必要）
        var ACTION_LOG_ENABLE = "net.yu_yum.LOG_ENABLE"
        var ACTION_LOG_DISABLE = "net.yu_yum.LOG_DISABLE"

        private var sEnabled: Boolean = false

        init {
            sEnabled = BuildConfig.DEBUG
        }

        fun d(tag: String, msg: String) {
            if (sEnabled) {
                Log.d(tag, msg)
            }
        }

        fun i(tag: String, msg: String) {
            if (sEnabled) {
                Log.i(tag, msg)
            }
        }

        fun w(tag: String, msg: String) {
            if (sEnabled) {
                Log.w(tag, msg)
            }
        }

        fun e(tag: String, msg: String) {
            if (sEnabled) {
                Log.e(tag, msg)
            }
        }
    }
}
